corehq/motech/repeaters/repeater_generators.py

```diff
@@ -739,3 +739,20 @@ def content_type(self):
 
     def get_payload(self, repeat_record, location):
         return json.dumps(location.to_json())
+
+
+class DataSourcePayloadGenerator(BasePayloadGenerator):
+    format_name = 'json'
+    format_label = _('JSON')
+
+    @property
+    def content_type(self):
+        return 'application/json'
+
+    def get_payload(self, repeat_record, transaction_log):
+        data = {
+            "data": transaction_log.row_data,
+            "action": transaction_log.action,
+            "data_source_id": transaction_log.data_source_id
```

Comment:
nit: not sure how this is intended to be used but how about adding a "row_id" or "doc_id" outside the data for easy fetching of the ID/row that needs to be updated?

Reply:
Agreed. The `row_id` is the `doc_id`. See [this](https://github.com/dimagi/commcare-hq/blob/63a9f23802a6c6d9571e628684274c1bec201c97/corehq/motech/repeaters/signals.py#L112). The reason for naming it `row_id`, is basically because it is the `row_id` of the Datasource table

Reply:
Cool. I meant within the json here, there could be a row_id property that the receiver could use directly instead of finding that in the data property.